mail-manager/src/main/scala/it/pagopa/interop/commons/mail/MailData.scala

```diff
@@ -1,23 +1,41 @@
 package it.pagopa.interop.commons.mail
 
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+import spray.json._
+import io.circe._
+import io.circe.generic.semiauto._
 import javax.mail.internet.InternetAddress
+import it.pagopa.interop.commons.utils.SprayCommonFormats.uuidFormat
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
 import org.typelevel.literally.Literally
 import scala.util.Try
 import courier._
+import java.util.UUID
 
 sealed trait Mail {
   val recipients: Seq[InternetAddress]
   val subject: String
 
   def renderContent: Content = this match {
-    case TextMail(_, _, body, Nil) => Text(body)
-    case TextMail(_, _, body, as)  =>
+    case TextMail(_, _, body, Nil)           => Text(body)
+    case TextMail(_, _, body, as)            =>
       as.foldLeft(Multipart().text(body)) { (c, a) => c.attachBytes(a.bytes, a.name, a.mimeType) }
-    case HttpMail(_, _, body, as)  =>
+    case HttpMail(_, _, body, as)            =>
       as.foldLeft(Multipart().html(body)) { (c, a) => c.attachBytes(a.bytes, a.name, a.mimeType) }
+    case InteropEnvelope(_, _, _, body, Nil) => Text(body)
+    case InteropEnvelope(_, _, _, body, as)  =>
+      as.foldLeft(Multipart().text(body)) { (c, a) => c.attachBytes(a.bytes, a.name, a.mimeType) }
   }
 }
 
+final case class InteropEnvelope(
```

Comment:
@beetlecrunch is there a reason why we are not using the generic `TextMail`?

Reply:
`InteropEnvelope` is sent to a FIFO queue, so we need a unique value to identify the object.
The main difference is the presence of the `id` field in the `InteropEnvelope`.
Maybe we can use only one model, but the `id` field should be part of it.
@galales what do you think?
In case we want to maintain both models, renaming InteropEnvelop with something like `CertifiedTextMail` or `UniqueTextMail` should be a good choice.

Reply:
The intent of the `id` field was also to trace the PEC mail flow in the logs.